integration-tests/src/test/java/org/zowe/apiml/integration/authentication/oauth2/OktaOauth2Test.java

```diff
@@ -64,7 +76,118 @@ void thenValidateReturns401() {
                 .then().statusCode(HttpStatus.SC_UNAUTHORIZED);
         }
+
+        @Test
+        void whenPassingToHeader_thenAddFailureHeader() {
+            RestAssured.useRelaxedHTTPSValidation();
+            given()
+                .contentType(ContentType.JSON)
+                .header(new Header(ApimlConstants.OIDC_HEADER_NAME, token))
+                .when()
+                .get(ZOWE_JWT_REQUEST_ENDPOINT)
+                .then().statusCode(200)
+                .body("headers", hasKey("x-zowe-auth-failure"))
+                .body("headers.x-zowe-auth-failure", is("ZWEAG103E The token has expired"))
+                .header(ApimlConstants.AUTH_FAIL_HEADER, startsWith("ZWEAG103E The token has expired"));
+        }
+    }
+
+    @Nested
+    class WhenTestingZoweJwtScheme {
+        private final String token = SecurityUtils.validOktaAccessToken();
+
+        @Test
+        void givenTokenInHeader_thenReturnInfo() {
+            RestAssured.useRelaxedHTTPSValidation();
+
+            given()
+                .contentType(ContentType.JSON)
+                .header(new Header(ApimlConstants.OIDC_HEADER_NAME, token))
+                .when()
+                .get(ZOWE_JWT_REQUEST_ENDPOINT)
+                .then().statusCode(200)
+                .body("headers", not(hasKey("x-zowe-auth-failure")))
+                .body("headers.cookie", isNotNull())
+                .body("cookies.apimlAuthenticationToken", isNotNull());
+        }
+    }
+
+    @Nested
+    class WhenTestingZosmfScheme {
+        private final String token = SecurityUtils.validOktaAccessToken();
+
+        @Test
+        void givenTokenInHeader_thenReturnInfo() {
+            RestAssured.useRelaxedHTTPSValidation();
+
+            given()
+                .contentType(ContentType.JSON)
+                .header(new Header(ApimlConstants.OIDC_HEADER_NAME, token))
+                .when()
+                .get(ZOSMF_REQUEST_ENDPOINT)
+                .then().statusCode(200)
+                .body("headers", not(hasKey("x-zowe-auth-failure")))
+                .body("headers.cookie", isNotNull())
+                .body("cookies.apimlAuthenticationToken", isNotNull());
+        }
+    }
+
+    @Nested
+    class WhenTestingSafIdtScheme {
+        private final String token = SecurityUtils.validOktaAccessToken();
+
+        @Test
+        void givenTokenInHeader_thenReturnInfo() {
+            RestAssured.useRelaxedHTTPSValidation();
+
+            given()
+                .contentType(ContentType.JSON)
+                .header(new Header(ApimlConstants.OIDC_HEADER_NAME, token))
+                .when()
+                .get(SAF_IDT_REQUEST_ENDPOINT)
+                .then().statusCode(200)
+                .body("headers", not(hasKey("x-zowe-auth-failure")))
+                .body("headers.cookie", isNotNull())
+                .body("cookies.apimlAuthenticationToken", isNotNull());
+        }
+    }
+
+    @Nested
+    class GivenOktaTokenRequest {
```

Comment:
how is this different from previous tests?

Reply:
Tests are redesigned to cover all combinations of valid x invalid tokens and schemas.